src/microbe_stage/OrganelleDefinition.cs

```diff
@@ -271,14 +271,23 @@ public enum OrganelleGroup
     /// <summary>
     ///   Gets the visual scene that should be used to represent this organelle (if there is one)
     /// </summary>
-    /// <para cref="modelInfo">
+    /// <param name="upgrades">
+    ///   Some upgrades alter organelle visuals
+    /// </param>
+    /// <param name="modelInfo">
     ///   The model info returned like this (as it may be a struct type this can't return a nullable reference without
     ///   boxing)
-    /// </para>
+    /// </param>
     /// <returns>True when this has a scene</returns>
-    public bool TryGetGraphicsScene(out LoadedSceneWithModelInfo modelInfo)
+    public bool TryGetGraphicsScene(OrganelleUpgrades? upgrades, out LoadedSceneWithModelInfo modelInfo)
     {
-        // TODO: allow this to be affected by upgrades (add as a first parameter to this method)
+        var hasUpgradeGraphics = TryGetGraphicsForUpgrade(upgrades, out var upgradeGraphics);
+
+        if (hasUpgradeGraphics)
```

Comment:
This doesn't need a separate bool so I'd recommend something like:

```c#
if (TryGetGraphicsForUpgrade(upgrades, out modelInfo)
{
     return true;
}
```

then also the line `modelInfo = default(LoadedSceneWithModelInfo);` should be removable to avoid one more data copy.

Reply:
No, just remove the line entirely, it should compile without it.